src/Microsoft.Framework.Logging/LoggerStructureFormat.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Framework.Logging
+{
+    public class LoggerStructureFormat : ILoggerStructure
+    {
+        private static ConcurrentDictionary<string, Formatter> _formatters = new ConcurrentDictionary<string, Formatter>();
+        private readonly Formatter _formatter;
+        private readonly object[] _values;
+
+        public LoggerStructureFormat(string format, object[] values)
+        {
+            _formatter = _formatters.GetOrAdd(format, _ => new Formatter(format));
```

Comment:
Curious - can you pass `format` in as the parameter to the lambda? Otherwise it looks like it's putting `format` in the closure, which I don't think is intentional (or needed) here. Not 100% sure about this though.